Library/OcBootManagementLib/VariableManagement.c

```diff
@@ -293,40 +357,16 @@ OcDeleteVariables (
   }
 
   if ((BootProtect & OC_BOOT_PROTECT_VARIABLE_BOOTSTRAP) != 0) {
-    BootOptionIndexSize = sizeof (BootOptionIndex);
-    Status = gRT->GetVariable (
-      OC_BOOTSTRAP_INDEX_VARIABLE_NAME,
-      &gOcVendorVariableGuid,
-      NULL,
-      &BootOptionIndexSize,
-      &BootOptionIndex
-      );
-    if (!EFI_ERROR (Status)) {
-      UnicodeSPrint (
-        BootOptionVariable,
-        sizeof (BootOptionVariable),
-        L"Boot%04x",
-        BootOptionIndex
-        );
-      
-      Status = GetVariable2 (
-        BootOptionVariable,
+    BootOption = InternalGetBootstrapBootData (&BootOptionSize);
+    if (BootOption != NULL) {
+      DEBUG ((
+        DEBUG_INFO,
+        "OCB: Found %g:%s for preservation of %u bytes\n",
         &gEfiGlobalVariableGuid,
-        &BootOption,
-        &BootOptionSize
-        );
-      if (!EFI_ERROR (Status)) {
-        DEBUG ((
-          DEBUG_INFO,
-          "OCB: Found %g:%s for preservation of %u bytes\n",
-          &gEfiGlobalVariableGuid,
-          BootOptionVariable,
-          (UINT32) BootOptionSize
-          ));
-      }
-    } 
-    
-    if (EFI_ERROR (Status)) {
+        BootOptionVariable,
```

Comment:
Hrm, why do we still have OC_BOOTSTRAP_INDEX_VARIABLE_NAME?

Reply:
Missed, thanks